src/bufr/IodaEncoder/IodaEncoder.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "IodaEncoder.h"
+
+#include <memory>
+
+#include "DataContainer.h"
+
+
+namespace Ingester
+{
+    IodaEncoder::IodaEncoder(const eckit::Configuration& conf) :
+        description_(IodaDescription(conf))
+    {
+    }
+
+    IodaEncoder::IodaEncoder(const IodaDescription& description) :
+        description_(description)
+    {
+    }
+
+    ioda::ObsGroup IodaEncoder::encode(const std::shared_ptr<DataContainer>& dataContainer,
+                                       bool append)
+    {
+        auto backendParams = ioda::Engines::BackendCreationParameters();
+
+        if (description_.getBackend() == ioda::Engines::BackendNames::Hdf5File)
+        {
+            backendParams.fileName = description_.getFilepath();
+        }
+
+        backendParams.openMode = ioda::Engines::BackendOpenModes::Read_Write;
+        backendParams.createMode = ioda::Engines::BackendCreateModes::Truncate_If_Exists;
+        backendParams.action = append ? ioda::Engines::BackendFileActions::Open : \
+                                        ioda::Engines::BackendFileActions::Create;
+        backendParams.flush = true;
+        backendParams.allocBytes = dataContainer->size();
+
+        auto rootGroup = ioda::Engines::constructBackend(description_.getBackend(), backendParams);
```

Comment:
Not sure if you know, @rmclaren , or if we need to ask @rhoneyager , but where do you define chunk size and compression/deflate level for the netCDF backend? The output file sees to be deflate level 6 and the chunk size is equal to the total dataset (2.8 million) which seems too large and inefficient. Can we set this at a per variable level in the ioda_engines like the netCDF C API allows? I'm mainly curious if we can reduce runtime by reducing deflate level.

Reply:
@CoryMartin-NOAA I took a look and you can set the chunk size and deflate level on a per variable basis like you suggest with the IodaEncoder, I just didn't give you a way to do it via the YAML configuration. I don't think it would be very difficult to add this however. I could add some optional parameters to the Ioda Variables where you could specify these. Would that make sense, or would it make more sense to hard code some reasonable number for both (or a combination of the two).

Reply:
YAML definitely, have a default option, but we should be able to set chunk size either explicitly or as a function of the variable size, and the deflate level can probably be the same for all variables.

Reply:
You now have the ability to specify the chunk size and compression level in the YAML file.